pkg/uploader/uploader.go

```diff
@@ -50,15 +53,49 @@ func (up *Uploader) Do(dstURL string) error {
 		return fmt.Errorf("invalid destination AMS URL ('%s'): %w", dstURL, err)
 	}
 	u.Path = path.Join(u.Path, "/sap/ams/v1/ams-instances/", up.AMSInstanceID, "/dcl-upload")
-	r, err := http.NewRequest(http.MethodPost, u.String(), buf)
+	resp, err := up.DoWithRetries(u.String(), body.Bytes(), maxRetries)
 	if err != nil {
-		return fmt.Errorf("could not create DCL upload request %w", err)
-	}
-	r.Header.Set("Content-Type", "application/gzip")
-	resp, err := up.Client.Do(r)
-	if err != nil {
-		return fmt.Errorf("DCL upload request unsuccessful: %w", err)
+		return fmt.Errorf("could not build upload request: %w", err)
 	}
 	defer resp.Body.Close()
 	return up.logResponse(resp)
 }
+
+const maxRetries = 9
+
+var RetryPeriod = 10 * time.Second
+
+func (up *Uploader) DoWithRetries(url string, body []byte, maxRetries int) (*http.Response, error) {
+	resp, err := up.do(url, body)
+	if err != nil {
+		return nil, fmt.Errorf("DCL upload request unsuccessful: %w", err)
+	}
+	retries := 0
+	for resp.StatusCode == http.StatusForbidden && retries < maxRetries {
```

Comment:
The AMS server responds with Unauthorized(401), if the certificate is not valid yet:
https://github.wdf.sap.corp/CPSecurity/cas-ams-server/blob/master/ams-server/api/auth/middleware.go#L55

Reply:
Whoops!